packages/nuxt/src/app/plugins/http-client-hints/http-client-hints.server.ts

```diff
@@ -0,0 +1,292 @@
+import type { IncomingHttpHeaders, ServerResponse } from 'node:http'
+import { readonly } from 'vue'
+import type { HttpClientHintsConfiguration, HttpClientHintsRequest } from './types'
+import { type Browser, parseUserAgent } from './detect-browser'
+import { NuxtHTTPClientHintsState } from './types'
+import { defineNuxtPlugin, useNuxtApp, useRuntimeConfig } from '#app/nuxt'
+import { useCookie } from '#app/composables/cookie'
+import { useState } from '#app/composables/state'
+import { useRequestEvent } from '#app/composables/ssr'
+
+export default defineNuxtPlugin<{ httpClientHints: HttpClientHintsRequest }>({
+  name: 'nuxt:http-client-hints:server',
+  enforce: 'pre',
+  setup () {
+    const event = useRequestEvent()
+    const httpClientHintsConfiguration = useRuntimeConfig().public.httpClientHints as HttpClientHintsConfiguration
+    const httpClientHints = useState<HttpClientHintsRequest>(NuxtHTTPClientHintsState)
+
+    const request = event.node.req
```

Comment:
For maximum future compatibility with non nodejs runtime, we should avoid relying on event.node and use h3 composables as much as possible

Reply:
Yeah, I just ported the logic, I need to review this part.

I'm going to extract browser detection and headers extraction to another repo  (later transfer it to unjs, maybe 2 separated repos)

Reply:
why is h3 using node?

![imagen](https://github.com/nuxt/nuxt/assets/6311119/8cd7da5d-36de-4605-af82-b4aeb02dff7c)

Reply:
done